core/src/AzureMcp.Core/Commands/ToolMetadata.cs

```diff
@@ -70,6 +70,25 @@ public sealed class ToolMetadata
     /// </remarks>
     public bool ReadOnly { get; init; } = false;
 
+    /// <summary>
+    /// Gets or sets whether this tool handles or returns secret or sensitive information.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// If <see langword="true"/>, the tool may handle or return sensitive information such as
+    /// credentials, API keys, connection strings, or other confidential data.
+    /// If <see langword="false"/>, the tool does not handle or return sensitive information.
+    /// </para>
+    /// <para>
+    /// This property helps MCP clients understand whether special handling or masking
+    /// may be required for the tool's inputs or outputs.
```

Comment:
>    This property helps MCP clients understand whether special handling or masking  may be required for the tool's inputs or outputs.

Regarding this documentation, my understanding is that we cannot return this flag to the "MCP client", since only flags or properties defined in the [`ModelContextProtocol.Protocol.ToolAnnotations`](https://github.com/modelcontextprotocol/csharp-sdk/blob/main/src/ModelContextProtocol.Core/Protocol/ToolAnnotations.cs) type (that maps to the spec that Jon linked) will be sent over the channel (stdio or wire) to the "MCP client" by the MCP SDK. If so, we may want to adjust the documentation. My understanding is that such additional flags could help us internally filter the visibility of tools (e.g., service start --disable-secrets-tools).

Reply:
You are right. We cannot add the hints into the response yet. Now we use it as the source of truth of the traits of the tools.

Reply:
Do you have background info on this change?  Like what is the plan, why are we doing this? What are next steps etc?  It would be good to include that info in PRs so we know why we are looking it.  Thanks

Reply:
My bad, updated the description of the PR.

It is for https://github.com/Azure/azure-mcp/issues/744 @jongio

Reply:
Before these hints are incorporated into the MCP spec, we will rely solely on the code as the source of truth for the tools' traits and may auto-generate documentation from it.

Reply:
As discussed with Josh:

We add it for a command line switch and for doc generation.

Reply:
@jongio please let me know if you have concerns.

Reply:
Currently, we use this metadata only to filter tools. If https://github.com/modelcontextprotocol/modelcontextprotocol/issues/711 is approved, we may update the tag to "sensitive."